src/geo/projection/globe_util.js

```diff
@@ -308,13 +322,50 @@ export function aabbForTileOnGlobe(tr: Transform, numTiles: number, tileId: Cano
     return new Aabb(cornerMin, cornerMax);
 }
 
-export function tileCornersToBounds({x, y, z}: CanonicalTileID): LngLatBounds {
-    const s = 1.0 / (1 << z);
-    const sw = new LngLat(lngFromMercatorX(x * s), latFromMercatorY((y + 1) * s));
-    const ne = new LngLat(lngFromMercatorX((x + 1) * s), latFromMercatorY(y * s));
+function tileCornersInMercator({x, y, z}: CanonicalTileID): [number, number, number, number] {
+    const tileScale = 1.0 / (1 << z);
+    const west = x * tileScale;
+    const east = west + tileScale;
+    const north = y * tileScale;
+    const south = north + tileScale;
+    return [south, west, north, east];
+}
+
+export function tileCornersToBounds(tileId: CanonicalTileID): LngLatBounds {
+    const [south, west, north, east] = tileCornersInMercator(tileId);
```

Comment:
Nit: since this is a hot path function, the goal of optimization here was to avoid redundant array allocation — first creating a 4-item array and then immediately destructuring and discarding it, adding to GC load. Can we revert to the previous `tileCornersToBounds` implementation and then inline `s/w/n/e` calculation in `mercatorTileCornersInCameraSpace`? Sometimes the benefit of reusing a chunk of code doesn't outweigh the cost.